src/main.rs

```diff
@@ -17,110 +17,76 @@
 /// If only the path is given, this will generate a summary report of any claims in that file
 /// If a manifest definition json file is specified, the claim will be added to any existing claims
 ///
-use std::fs::{create_dir_all, remove_dir_all};
-use std::path::{Path, PathBuf};
+use std::fs::{create_dir_all, remove_dir_all, File};
+use std::io::Write;
+use std::path::PathBuf;
 
 use anyhow::{anyhow, bail, Context, Result};
-use c2pa::{Error, ManifestStore, ManifestStoreReport};
-use structopt::{clap::AppSettings, StructOpt};
+use c2pa::{Error, Ingredient, Manifest, ManifestStore, ManifestStoreReport};
+use clap::{AppSettings, Parser};
 
 mod info;
 use info::info;
-pub mod manifest_config;
-use manifest_config::ManifestConfig;
-mod fs_report;
 mod signer;
-use signer::get_c2pa_signer;
+use signer::SignConfig;
 
-// define the command line options
-#[derive(Debug, StructOpt)]
-#[structopt(about = "Tool for displaying and creating C2PA manifests.",global_settings = &[AppSettings::ColoredHelp, AppSettings::ArgRequiredElseHelp])]
+/// Tool for displaying and creating C2PA manifests."
+#[derive(Parser, Debug)]
+#[clap(author, version, about, long_about = None, setting = AppSettings::ArgRequiredElseHelp)]
```

Comment:
:+1: for transition to clap. Using it for a personal (not open-source) project and liking it.